packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -29,71 +29,124 @@ import {
 	BlockControls,
 	InnerBlocks,
 	InspectorControls,
-	URLPopover,
+	__experimentalLinkControl as LinkControl,
 } from '@wordpress/block-editor';
-import {
-	Fragment,
-	useRef,
-	useState,
-} from '@wordpress/element';
+import { Fragment, useState, useRef, useEffect } from '@wordpress/element';
 
 function NavigationMenuItemEdit( {
 	attributes,
 	isSelected,
 	isParentOfSelectedBlock,
 	setAttributes,
+	fetchSearchSuggestions,
 } ) {
-	const plainTextRef = useRef( null );
+	const { label, opensInNewTab, title, url } = attributes;
+	const link = title ? { title, url } : null;
```

Comment:
@talldan I'm thinking we could (_not_ in this PR) look to extract some of this state into a custom hook that resides within the `link-control` directory. What do you think? I can imagine this being repeated ad infinitum.